src/testsuite/vec_int512_dummy.c

```diff
@@ -0,0 +1,743 @@
+  mpx6 = vec_addeq (&mq, mpx6, mq5, mc);
+  COMPILE_FENCE;
+  mpx7 = vec_muludq (&mq7, m1.vx7, m2);
+  mpx7 = vec_addeq (&mc, mpx7, mq6, mq);
+  COMPILE_FENCE;
+
+  mpx8 = vec_muludq (&mq8, m1.vx8, m2);
+  mpx8 = vec_addeq (&mq, mpx8, mq7, mc);
+  COMPILE_FENCE;
+  mpx9 = vec_muludq (&mq9, m1.vx9, m2);
+  mpx9 = vec_addeq (&mc, mpx9, mq8, mq);
+  COMPILE_FENCE;
+  mpx10 = vec_muludq (&mq10, m1.vx10, m2);
+  mpx10 = vec_addeq (&mq, mpx10, mq9, mc);
+  COMPILE_FENCE;
+  mpx11 = vec_muludq (&mq11, m1.vx11, m2);
+  mpx11 = vec_addeq (&mc, mpx11, mq10, mq);
+  COMPILE_FENCE;
+
+  mpx12 = vec_muludq (&mq12, m1.vx12, m2);
+  mpx12 = vec_addeq (&mq, mpx12, mq11, mc);
+  COMPILE_FENCE;
+  mpx13 = vec_muludq (&mq13, m1.vx13, m2);
+  mpx13 = vec_addeq (&mc, mpx13, mq12, mq);
+  COMPILE_FENCE;
+  mpx14 = vec_muludq (&mq14, m1.vx14, m2);
+  mpx14 = vec_addeq (&mq, mpx14, mq13, mc);
+  COMPILE_FENCE;
+  mpx15 = vec_muludq (&mq15, m1.vx15, m2);
+  mpx15 = vec_addeq (&mc, mpx15, mq14, mq);
+  mq15 = vec_adduqm (mc, mq15);
+  COMPILE_FENCE;
+
+  result.vx0 = mpx0;
+  result.vx1 = mpx1;
+  result.vx2 = mpx2;
+  result.vx3 = mpx3;
+  result.vx4 = mpx4;
+  result.vx5 = mpx5;
+  result.vx6 = mpx6;
+  result.vx7 = mpx7;
+  result.vx8 = mpx8;
+  result.vx9 = mpx9;
+  result.vx10 = mpx10;
+  result.vx11 = mpx11;
+  result.vx12 = mpx12;
+  result.vx13 = mpx13;
+  result.vx14 = mpx14;
+  result.vx15 = mpx15;
+  result.vx16 = mq15;
+  return result;
+}
+
+__VEC_U_1024
+__VEC_PWR_IMP (test_mulu512x512) (__VEC_U_512 m1, __VEC_U_512 m2)
+{
+#if 1
+  return vec_mul512x512_inline (m1, m2);
+#else
+  __VEC_U_1024 result;
+  vui128_t mc, mp, mq;
+  __VEC_U_640 mp3, mp2, mp1, mp0;
+#if 1
+
+  mp0 = test_mul512x128_inline (m1, m2.vx0);
+  result.vx0 = mp0.vx0;
+  result.vx1 = mp0.vx1;
+  result.vx2 = mp0.vx2;
+  result.vx3 = mp0.vx3;
+  result.vx4 = mp0.vx4;
+  COMPILE_FENCE;
+  mp1 = test_mul512x128_inline (m1, m2.vx1);
+  result.vx1 = vec_addcq (&mq, mp1.vx0, result.vx1);
+  result.vx2 = vec_addeq (&mp, mp1.vx1, result.vx2, mq);
+  result.vx3 = vec_addeq (&mq, mp1.vx2, result.vx3, mp);
+  result.vx4 = vec_addeq (&mp, mp1.vx3, result.vx4, mq);
+  result.vx5 = vec_addcq (&result.vx6, mp1.vx4, mp);
+  COMPILE_FENCE;
+  mp2 = test_mul512x128_inline (m1, m2.vx2);
+  result.vx2 = vec_addcq (&mq, mp2.vx0, result.vx2);
+  result.vx3 = vec_addeq (&mp, mp2.vx1, result.vx3, mq);
+  result.vx4 = vec_addeq (&mq, mp2.vx2, result.vx4, mp);
+  result.vx5 = vec_addeq (&mp, mp2.vx3, result.vx5, mq);
+  result.vx6 = vec_addeq (&result.vx7, mp2.vx4, result.vx6, mp);
+  COMPILE_FENCE;
+  mp3 = test_mul512x128_inline (m1, m2.vx3);
+  result.vx3 = vec_addcq (&mq, mp3.vx0, result.vx3);
+  result.vx4 = vec_addeq (&mp, mp3.vx1, result.vx4, mq);
+  result.vx5 = vec_addeq (&mq, mp3.vx2, result.vx5, mp);
+  result.vx6 = vec_addeq (&mp, mp3.vx3, result.vx6, mq);
+  result.vx7 = vec_addeuqm (result.vx7, mp3.vx4, mp);
+#else
+  mp0 = test_muluq512x128_inline_PWR9 (m1, m2.vx0);
```

Comment:
this is all `#ifdef`d out, as is more below.  Do you want to keep it all?

Reply:
I removed the test_muluq512x128_inline_PWR9 based section. This was a dead end. But I will keep the rest. 

This is how I try ideas.